modules/aws_asg/high-cpu-check.bash

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+set -e
+set -o pipefail
+
+__logger() {
+  level="${1}" && shift
+  msg="${1}" && shift
+  date="$(date --iso-8601=seconds)"
+  log_msg="tag=cron time=${date} level=${level} msg=\"${msg}\""
+  for bit in "${@}"; do
+    log_msg="${log_msg} ${bit}"
+  done
+
+  logger -t "$(basename "${0}")" "${log_msg}"
+}
+
+__die() {
+  local status="${1}"
+  local code="${2}"
+  __logger "info" \
+    "cron finished" \
+    "status=${status}"
+  exit "${code}"
+}
+
+report_greedy_containers() {
+  max_cpu="${1}"
+  : "${RUNDIR:=/var/tmp/travis-run.d}"
+  instance_id="$(cat "${RUNDIR}/instance-id")"
+  instance_ip="$(cat "${RUNDIR}/instance-ipv4")"
+
+  IFS=$'\n'
+  stats="$(docker stats --no-stream --format "{{.Container}} {{.CPUPerc}} {{.Name}}" | tr -d '%')"
+
+  echo "${stats}" | while IFS=" " read -r cid usage_as_float name; do
```

Comment:
:grin: I don't understand why the two-step of fetching stats and then `echo`'ing stats is needed :heart: 

Does this not work?

```
docker stats --no-stream --format "{{.Container}} {{.CPUPerc}} {{.Name}}" | while read -r cid usage_as_float name; do
  usage_as_float="${usage_as_float//%/}"
  # ... and everything else can stay as-is, AFAICT
```

Reply:
Purely for readability!

Reply:
:thinking: mmmm I'm sorry I'm stuck on this.  Defining a large-ish string variable and then `echo`'ing and, then checking if it's empty to indicate a counter ... It feels pretty brittle to me :confused: 

I understand if you want to get this merged as-is.  I'll stop commenting and get out of the way :heart: